### ISSUE TITLE
Tilemap light occluders affect all open scenes in the editor

### ISSUE THREAD

**Issue by pmoosi** [godotengine/godot#89769]
### Tested versions

- Reproducible in v4.3.dev.mono.custom_build [fe01776f0] and v4.2.2.rc.mono.custom_build [15c5e10ea]
- Not reproducible in v4.2.1.stable.mono.official [b09f793f5]

### System information

Godot v4.2.2.rc.mono (15c5e10ea) - Ubuntu 22.04.4 LTS 22.04 - Wayland - Vulkan (Forward+) - integrated AMD Unknown (RADV RENOIR) () - AMD Ryzen 7 5800H with Radeon Graphics (16 Threads)

### Issue description

When adding a tile from a tileset with a light occluder to a scene, it affects all other open scenes in the editor. That is, the light is blocked on all scenes. It seems to be only the case with tiles and not other occluders.

![Scene1](https://github.com/godotengine/godot/assets/25228699/ea3de384-8877-4c9f-80e0-ab6cacf580ad)
![Scene2](https://github.com/godotengine/godot/assets/25228699/d22d6aca-78d8-4073-99a3-8169c2f57dcd)

### Steps to reproduce

In the MRP open Scene1.tscn and Scene2.tscn. The light-blocking tiles from Scene1 also block the light in Scene2 in the editor.

### Minimal reproduction project (MRP)

[mrp.zip](https://github.com/godotengine/godot/files/14716151/mrp.zip)

**Comment by Macksaur** [comment]
@groud @AThousandShips After a bisect fe203d700333039fb583dfd2e01f754c70ba1486 seems to be the cause of this. Not sure what's going on, does the logic in `TileMapLayer::_rendering_update`, etc., rely on specifically being called in a deferred manner? i.e. after the node has been fully removed from the tree? It looks to be skipping the removal of occluders amongst other things.

**Comment by AThousandShips** [comment]
Will take a look later today, not sure exactly here could be some issue with how the data was handled before that was exposed, nothing really changed about these things with my PR in the details of updating this just how it was processed, might be a missing check somewhere
